src/integrators/type.jl

```diff
@@ -101,8 +101,10 @@ mutable struct ODEIntegrator{algType<:OrdinaryDiffEqAlgorithm,IIP,uType,tType,pT
   saveiter::Int
   saveiter_dense::Int
   cache::CacheType
-  callback_cache::Array{uType}
-  previous_condition::Array{uType}
+  callback_cache::CallbackCacheType
+  previous_condition::CallbackCacheType
+  callback_next_sign::CallbackCacheType
```

Comment:
This can be a bitarray?

Reply:
Yeah this can be. This was defaulted to float previously thats why I chose float. Lets change this to BitArray.